src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -4706,17 +4706,13 @@ def get_k8s_support_plan(self) -> Union[str, None]:
 
         :return: string or None
         """
-        # default to None
-        support_plan = None
-        # try to read the property value corresponding to the parameter from the `mc` object
-        if self.mc and hasattr(self.mc, "support_plan") and self.mc.support_plan is not None:
-            support_plan = self.mc.support_plan
-
-        # if specified by customer, use the specified value
+        # take input
         support_plan = self.raw_param.get("k8s_support_plan")
+        if support_plan is None:
+            # user didn't update this property, load from existing ManagedCluster
+            if self.mc and hasattr(self.mc, "support_plan") and self.mc.support_plan is not None:
+                support_plan = self.mc.support_plan
```

Comment:
I would like to confirm if this change of default behavior will cause a breaking change for customers? If so, please add the `BREAKING CHANGE` label to the PR title

Reply:
It is formally a breaking change, but the previous behavior was not as expected and will cause users who have enabled support plan to encounter problems when using the `az aks update` command, so I think this change is a just bug fix.

Reply:
1. this --k8s-support-plan have no major adoption yet, otherwise this would have been reported.
2. yes, as Fuming mentioned, this is fixing unexpected behavior.

Reply:
Got it, thanks!